datafusion/core/src/datasource/physical_plan/parquet/page_filter.rs

```diff
@@ -164,17 +169,23 @@ impl PagePruningPredicate {
         let mut row_selections = Vec::with_capacity(page_index_predicates.len());
         for predicate in page_index_predicates {
             // find column index by looking in the row group metadata.
-            let col_idx = find_column_index(predicate, &groups[0]);
 
+            let name = find_column_name(predicate);
+            let mut parquet_col = None;
+            if name.is_some() {
+                parquet_col =
+                    parquet_column(parquet_schema, arrow_schema, name.unwrap().as_str());
+            }
```

Comment:
I think a more rust ideomatic way of doing this pattern that avoids the unwrap and mut is something like

```suggestion
            let parquet_col = name
              .and_then(|name| {
                    parquet_column(parquet_schema, arrow_schema, name.as_str());
            });
```

Reply:
aah thank you , i was searching for something like this. I will add this to my toolbox.